pkg/tss/tpm.go

```diff
@@ -10,27 +12,16 @@ import (
 	"io/ioutil"
 )
 
-// MatchesConfig returns true if the TPM satisfies the constraints
-// specified by the given config.
-func (t *probedTPM) matchesConfig(config OpenConfig) bool {
-	return config.TPMVersion == TPMVersionAgnostic || t.Version == config.TPMVersion
-}
-
 // OpenTPM initializes access to the TPM based on the
 // config provided.
-func OpenTPM(config *OpenConfig) (*TPM, error) {
-	if config == nil {
-		config = &OpenConfig{}
-	}
+func OpenTPM() (*TPM, error) {
 	candidateTPMs, err := probeSystemTPMs()
 	if err != nil {
 		return nil, err
 	}
 
 	for _, tpm := range candidateTPMs {
-		if tpm.matchesConfig(*config) {
-			return openTPM(tpm)
-		}
+		return openTPM(tpm)
```

Comment:
check error? Otherwise move on to next candidate?

Reply:
good point